numpy/lib/polynomial.py

```diff
@@ -898,6 +898,8 @@ def polydiv(u, v):
     scale = 1. / v[0]
     q = NX.zeros((max(m - n + 1, 1),), w.dtype)
     r = u.copy()
+    if iscomplex(u).any() or iscomplex(v).any():
+        r=r.astype(complex)
```

Comment:
Just use `r = u.astype(w.dtype)` above

Reply:
But that wouldn't work if `u[0]` and `v[0]` are not complex, but other elements are complex.

Reply:
If `issubdtype(u.dtype, np.complexfloating)`, then `w.dtype` will be too

Reply:
Is there an issue with `issubdtype()` ? It returns True only if both the dtypes are same. Could you please confirm it?

Reply:
Not if you pass a type as the second argument like `np.complexfloating`? Can you give a failing example?

Reply:
Also, you don't need a condition here at all - i was just using `issubdtype` as an explanation tool.

Reply:
`issubdtype(np.float32,np.float64)` returns `False`. Same with result `issubdtype(np.float64,np.complexfloating)`.

Reply:
By design. np.float32 and np.float64 are different types. They are both subtypes of `np.floating`. `np.float64` is not a subtype of complex numbers, so that result is expected too.  `issubdtype(np.float64, np.inexact)` returns `True`, as does `issubdtype(np.float64, np.floating)`